contrib/raftexample/raft.go

```diff
@@ -77,27 +85,143 @@ func newRaftNode(id int, peers []string, join bool, proposeC <-chan string,
 		peers:       peers,
 		join:        join,
 		waldir:      fmt.Sprintf("raftexample-%d", id),
+		snapDir:     fmt.Sprintf("raftexample-%d-snap", id),
 		raftStorage: raft.NewMemoryStorage(),
+		kv:          kv,
```

Comment:
`raftNode` shouldn't know about `kv`; `raftNode` should only know about raft stuff